### PR TITLE
Tracking sheet performance tweaks

### PR DESCRIPTION
**Pull request by awisema**

### CODE REVIEW

File: app/models/grda_warehouse/hud/client.rb [hunk 1]
```diff
@@ -787,6 +787,27 @@ def self.disabled_client_ids
       disabled_client_scope.pluck(:id)
     end
 
+    def self.disabling_condition_client_scope
+      mre_t = Arel::Table.new(:most_recent_enrollments)
+      join = she_t.join(mre_t).on(she_t[:id].eq(mre_t[:current_id]))
```
Comment (eanders) [inline app/models/grda_warehouse/hud/client.rb:792]: Does this join need to include DateDeleted is null?

File: app/models/grda_warehouse/hud/client.rb [hunk 2]
```diff
@@ -787,6 +787,27 @@ def self.disabled_client_ids
...
+
+      where(
+        id: GrdaWarehouse::ServiceHistoryEnrollment.
+        with(
+          most_recent_enrollments:
+            GrdaWarehouse::ServiceHistoryEnrollment.
+              joins(:enrollment).
+              define_window(:client_by_update).partition_by(:client_id, order_by: { e_t[:DateUpdated] => :desc }).
+              select_window(:first_value, she_t[:id], over: :client_by_update, as: :current_id).
+              # where(project_type: GrdaWarehouse::Hud::Project::RESIDENTIAL_PROJECT_TYPE_IDS).
```
Comment (eanders) [inline app/models/grda_warehouse/hud/client.rb:802]: Should this be commented out?

Comment (eanders) [reply app/models/grda_warehouse/hud/client.rb:802]: Ah, yes, it probably should.

File: app/models/grda_warehouse/hud/client.rb (hunk 1 above)
Comment (awisema) [reply app/models/grda_warehouse/hud/client.rb:792]: This is by id to the window table (`most_recent_enrollments`) so it shouldn't contain any deleted enrollments as long as they are excluded in the window.

**Pull request merged**